include/zephyr/drivers/otp.h

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2025 STMicroelectronics
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @ingroup otp_interface
+ * @brief Main header file for OTP driver API.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_OTP_H_
+#define ZEPHYR_INCLUDE_DRIVERS_OTP_H_
+
+/**
+ * @brief Interfaces for One Time Programmable (OTP) Memory.
+ * @defgroup otp_interface One Time Programmable Memory
+ * @since 4.4
+ * @version 1.0.0
```

Comment:
API should be experimental at first? :slightly_smiling_face: 
```suggestion
 * @version 0.1.0
```